Slim/Utils/Scanner/Local/Async.pm

```diff
@@ -150,10 +150,7 @@ sub find {
 
 		$count++;
 
-		# XXX Not sure why, but sometimes there is no cached stat data available?!
-		if ( !(stat _)[9] ) {
```

Comment:
This check, as the comment suggests, was added specifically for cases where `_` was not defined. Are you saying that `(stat _)[9]` would be defined, but element 7 would be zero (or empty)? Maybe we should check for that value instead, then? I'm hesitating to remove the check, as re-reading the stats could potentially hurt performance. This might not be much of an issue on modern SSD based systems, but very well on eg. a Raspberry Pi with a slowish USB stick.

Reply:
I don't know if this has been discussed in the forums. But maybe @nyanloutre who reported the issue does?

> Are you saying that (stat _)[9] would be defined, but element 7 would be zero (or empty)?

Exactly. The thing that I don't understand is: for `stat _` to be cached, afaiu, it should have been called explicitly before. But I cannot find this initial call anywhere 🤔. Do you know where is that initial explicit call done?

My gut feeling is that for some reason that cached `stat` call concerns another file than the one we're interested in. And because it is non empty for most people, they don't see any issue.  In my case it is very often empty, maybe because it has been called on a directory or smt like that? Just an idea.

I'll try figure out what is actually going on next week. But in the meantime feel free to comment if you have any idea on how to debug the issue.

Reply:
I'm wondering whether the magic happens in `$file = File::Spec->canonpath($file);`. Looking at that code there seems to be a C version for some platforms, and a pure Perl version for others. The latter AFAIU would NOT stat the file. The C code I unfortunately don't understand well enough 🙈. But as the problem seems to be specific to NixOS (that's what you're using too, isn't it?), I was wondering whether there was no binary for it. Or whether NixOS handled files slightly differently to other platforms?

Anyway: as the existing code seems to be working for 99.99 (and probably some) percent of the users, I'd rather see an additional check for the file size to be > 0 than removing the test.

Reply:
Ok thanks.

It sounds like a solid lead. I'll have a look at the implementation of that function and figure out what is going on.